Missing break in Status::code_to_string switch case

Missing break in Status Switch case

https://github.com/TileDB-Inc/TileDB/blob/d3aeffaadedf0a126bf8809428b62a4e44deb1ff/tiledb/sm/misc/status.cc#L193

```
/tmp/TileDB/tiledb/sm/misc/status.cc: In member function ‘std::__cxx11::string tiledb::sm::Status::code_to_string() const’:
/tmp/TileDB/tiledb/sm/misc/status.cc:193:12: warning: this statement may fall through [-Wimplicit-fallthrough=]
       type = "[TileDB::Encryption] Error";
       ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/TileDB/tiledb/sm/misc/status.cc:194:5: note: here
     case StatusCode::Array:
     ^~~~
```

--- a/test/src/unit-capi-array.cc
+++ b/test/src/unit-capi-array.cc
@@ -226,13 +226,9 @@ TEST_CASE_METHOD(
   std::string array_name = FILE_URI_PREFIX + FILE_TEMP_DIR + "array_uri";
   create_temp_dir(FILE_URI_PREFIX + FILE_TEMP_DIR);
 
-  // Create context
-  int rc = tiledb_ctx_alloc(nullptr, &ctx_);
-  REQUIRE(rc == TILEDB_OK);
-
   // Create array
   tiledb_array_t* array;
-  rc = tiledb_array_alloc(ctx_, array_name.c_str(), &array);
+  int rc = tiledb_array_alloc(ctx_, array_name.c_str(), &array);
   REQUIRE(rc == TILEDB_OK);
 
   // Get URI when array is not opened (should not error)
@@ -264,4 +260,12 @@ TEST_CASE_METHOD(
   tiledb_array_free(&array);
 
   remove_temp_dir(FILE_URI_PREFIX + FILE_TEMP_DIR);
+}
+
+TEST_CASE_METHOD(
+    ArrayFx, "C API: Set null URI", "[capi], [array], [array-null-uri]") {
+  // Create context
+  tiledb_array_t* array;
+  int rc = tiledb_array_alloc(ctx_, nullptr, &array);
+  CHECK(rc == TILEDB_ERR);
 }

--- a/tiledb/sm/misc/status.cc
+++ b/tiledb/sm/misc/status.cc
@@ -191,6 +191,7 @@ std::string Status::code_to_string() const {
       break;
     case StatusCode::Encryption:
       type = "[TileDB::Encryption] Error";
+      break;
     case StatusCode::Array:
       type = "[TileDB::Array] Error";
       break;

--- a/tiledb/sm/misc/uri.cc
+++ b/tiledb/sm/misc/uri.cc
@@ -52,6 +52,14 @@ URI::URI() {
   uri_ = "";
 }
 
+URI::URI(char* path)
+    : URI((path == nullptr) ? std::string("") : std::string(path)) {
+}
+
+URI::URI(const char* path)
+    : URI((path == nullptr) ? std::string("") : std::string(path)) {
+}
+
 URI::URI(const std::string& path) {
   if (path.empty())
     uri_ = "";

--- a/tiledb/sm/misc/uri.h
+++ b/tiledb/sm/misc/uri.h
@@ -56,6 +56,22 @@ class URI {
    * @param path String that gets converted into an absolute path and stored
    *     as a URI.
    */
+  explicit URI(const char* path);
+
+  /**
+   * Constructor.
+   *
+   * @param path String that gets converted into an absolute path and stored
+   *     as a URI.
+   */
+  explicit URI(char* path);
+
+  /**
+   * Constructor.
+   *
+   * @param path String that gets converted into an absolute path and stored
+   *     as a URI.
+   */
   explicit URI(const std::string& path);
 
   /** Destructor. */

